nntrainer/tensor/cpu_backend/ggml_interface/nntr_ggml_impl/nntr_ggml_impl.cpp

```diff
@@ -9526,72 +9524,70 @@ void nntr_gemv_q4_0_8x8_q8_0(int n, float *__restrict s, size_t bs,
 
 #if !((defined(_MSC_VER)) && !defined(__clang__)) && defined(__aarch64__)
 #if defined(__ARM_FEATURE_SVE)
-  if (ggml_cpu_has_sve() && ggml_cpu_get_sve_cnt() == QK8_0) {
-    const void *b_ptr = vx;
-    const void *a_ptr = vy;
-    float *res_ptr = s;
-
-    __asm__ __volatile__(
-      "ptrue p0.b\n"
-      "add %x[b_ptr], %x[b_ptr], #0x10\n"
-      "1:" // Column loop
-      "add x22, %x[a_ptr], #0x2\n"
-      "mov z31.b, #0x0\n"
-      "mov x21, %x[nb]\n"
-      "2:" // Block loop
-      "ld1b { z30.b }, p0/Z, [%x[b_ptr]]\n"
-      "ld1b { z29.b }, p0/Z, [%x[b_ptr], #1, MUL VL]\n"
-      "mov z28.s, #0x0\n"
-      "mov z27.s, #0x0\n"
-      "ld1rd { z26.d }, p0/Z, [x22]\n"
-      "ld1b { z25.b }, p0/Z, [%x[b_ptr], #2, MUL VL]\n"
-      "sub x20, x22, #0x2\n"
-      "sub x21, x21, #0x1\n"
-      "ld1b { z24.b }, p0/Z, [%x[b_ptr], #3, MUL VL]\n"
-      "ld1rd { z23.d }, p0/Z, [x22, #8]\n"
-      "lsl z22.b, z30.b, #0x4\n"
-      "lsl z16.b, z29.b, #0x4\n"
-      "and z30.b, z30.b, #0xf0\n"
-      "and z29.b, z29.b, #0xf0\n"
-      "ld1rd { z21.d }, p0/Z, [x22, #16]\n"
-      "ld1rd { z20.d }, p0/Z, [x22, #24]\n"
-      "lsl z19.b, z25.b, #0x4\n"
-      "and z25.b, z25.b, #0xf0\n"
-      "ld1rh { z17.h }, p0/Z, [x20]\n"
-      "ld1h { z18.s }, p0/Z, [%x[b_ptr], #-1, MUL VL]\n"
-      "sdot z28.s, z22.b, z26.b\n"
-      "sdot z27.s, z16.b, z26.b\n"
-      "lsl z16.b, z24.b, #0x4\n"
-      "add x22, x22, #0x22\n"
-      "and z24.b, z24.b, #0xf0\n"
-      "add %x[b_ptr], %x[b_ptr], #0x90\n"
-      "fcvt z17.s, p0/m, z17.h\n"
-      "fcvt z18.s, p0/m, z18.h\n"
-      "sdot z28.s, z19.b, z23.b\n"
-      "sdot z27.s, z16.b, z23.b\n"
-      "fmul z18.s, z18.s, z17.s\n"
-      "sdot z28.s, z30.b, z21.b\n"
-      "sdot z27.s, z29.b, z21.b\n"
-      "sdot z28.s, z25.b, z20.b\n"
-      "sdot z27.s, z24.b, z20.b\n"
-      "uzp1 z17.s, z28.s, z27.s\n"
-      "uzp2 z16.s, z28.s, z27.s\n"
-      "add z17.s, z17.s, z16.s\n"
-      "asr z17.s, z17.s, #0x4\n"
-      "scvtf z17.s, p0/m, z17.s\n"
-      "fmla z31.s, p0/M, z17.s, z18.s\n"
-      "cbnz x21, 2b\n"
-      "sub %x[nc], %x[nc], #0x8\n"
-      "st1w { z31.s }, p0, [%x[res_ptr]]\n"
-      "add %x[res_ptr], %x[res_ptr], #0x20\n"
-      "cbnz %x[nc], 1b\n"
-      : [b_ptr] "+&r"(b_ptr), [res_ptr] "+&r"(res_ptr), [nc] "+&r"(nc)
-      : [a_ptr] "r"(a_ptr), [nb] "r"(nb)
-      : "memory", "p0", "x20", "x21", "x22", "z16", "z17", "z18", "z19", "z20",
-        "z21", "z22", "z23", "z24", "z25", "z26", "z27", "z28", "z29", "z30",
-        "z31");
-    return;
-  }
+  const void *b_ptr = vx;
```

Comment:
I really don't like seeing this kind of #ifdef / #endif in C code.

Let's not modify this in this PR, but later this year, please consider refactoring:
1. (best) identify micro-arch at run-time/launch-time, do "if", not "#if/#ifdef".
2. (not too bad. bearable but worse than 1.) no micro/macro arch #if/#ifdef in .c/cpp files, do that only in header files.